autoload/codefmt.vim

```diff
@@ -156,6 +157,65 @@ if !exists('s:gofmt')
   call codefmtlib#AddDefaultFormatter(s:gofmt)
 endif
 
+" Formatter: autopep8
+if !exists('s:autopep8')
+  let s:autopep8 = {
+      \ 'name': 'autopep8',
+      \ 'setup_instructions': 'Install autopep8 ' .
+          \ '(https://pypi.python.org/pypi/autopep8/).'}
+
+  function s:autopep8.IsAvailable() abort
+    return executable(s:plugin.Flag('autopep8_executable'))
+  endfunction
+
+  function s:autopep8.AppliesToBuffer() abort
+    return &filetype is# 'python'
+  endfunction
+
+  ""
+  " Reformat the current buffer with autopep8 or the binary named in
+  " @flag(autopep8_executable), only targeting the range between {startline} and
+  " {endline}.
+  " @throws ShellError
+  function s:autopep8.FormatRange(startline, endline) abort
+    " Hack range formatting by formatting range individually, ignoring context.
+    let l:executable = s:plugin.Flag('autopep8_executable')
+    if !exists('s:autopep8_supports_range')
+      let l:version_output =
+          \ maktaba#syscall#Create([l:executable, '--version']).Call().stderr
+      let s:autopep8_supports_range =
+          \ matchlist(l:version_output, '\m\Cautopep8 \(\d\+\)\.')[1] >= 1
+    endif
+
+    call maktaba#ensure#IsNumber(a:startline)
+    call maktaba#ensure#IsNumber(a:endline)
+    let l:lines = getline(1, line('$'))
+
+    if s:autopep8_supports_range
```

Comment:
Didn't know what to make of it. I thought it was similar to the bash things
which say "all arguments after this one are files" which confused me and
sent me off to read the docs.

Matt
On Jan 26, 2015 11:51 AM, "David Barnett" <notifications@github.com> wrote:

> In autoload/codefmt.vim
> <https://github.com/google/vim-codefmt/pull/4#discussion_r23544410>:
>
> > +  " @throws ShellError
> > +  function s:autopep8.FormatRange(startline, endline) abort
> > +    " Hack range formatting by formatting range individually, ignoring context.
> > +    let l:executable = s:plugin.Flag('autopep8_executable')
> > +    if !exists('s:autopep8_supports_range')
> > +      let l:version_output =
> > +          \ maktaba#syscall#Create([l:executable, '--version']).Call().stderr
> > +      let s:autopep8_supports_range =
> > +          \ matchlist(l:version_output, '\m\Cautopep8 \(\d\+\)\.')[1] >= 1
> > +    endif
> > +
> > +    call maktaba#ensure#IsNumber(a:startline)
> > +    call maktaba#ensure#IsNumber(a:endline)
> > +    let l:lines = getline(1, line('$'))
> > +
> > +    if s:autopep8_supports_range
>
> I didn't find it too confusing, but it could be more obvious at a glance
> if the lines were broken up differently:
>
>       let l:cmd = [
>           \ l:executable,
>           \ '--range', ''.a:startline, ''.a:endline,
>           \ '-' ]
>
> Matt, you still think a comment would be necessary like that? Was it more
> that you didn't notice the '-' or just didn't know what to make of it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/vim-codefmt/pull/4/files#r23544410>.
>